### ISSUE TITLE
How can I increase the Max context length（max token length） to 16K？

### ISSUE THREAD

**Issue by ANYMS-A** [meta-llama/llama3#219]
Hey, I would like to know how should I increase the maximum context length of the Llama3 model from 8K to 16K. 

The essential question is: what decide the model's max context length to 8K? IF my understanding is correct, I can increase the max context length as long as I have enough GPU memory and computation resources (and maybe time) right? 
Or is the 8K length related with the training data of the model?(i.e. the max length of the training data is up to 8K)

If I increase the max context length to 16K from 8K, by only changing the model's initialization argument, should I do a further finetune for the model with longer data sequence?
Will there be a performance degradation(except inference speed)if I do not apply any further finetune after increasing the max context length.

Thanks!

**Comment by dongwang218** [comment]
The context length can be increased by changing the sequence length parameter. However, the pretrained model is trained using text upto 8k tokens, just increase the context length will generate poor results. To generalize beyond 8k, further fine tuning with longer texts are necessary. There are documented steps to increase to >1M tokens such as this [article](https://gradient.ai/blog/scaling-rotational-embeddings-for-long-context-language-models), which also involves increasing the rope theta while increasing length of the training sequence. This example training [script](https://github.com/jzhang38/EasyContext/blob/main/train_scripts/EasyContext-1M-Llama-2-7B.sh) might also be helpful.

**Comment by ANYMS-A** [comment]
> The context length can be increased by changing the sequence length parameter. However, the pretrained model is trained using text upto 8k tokens, just increase the context length will generate poor results. To generalize beyond 8k, further fine tuning with longer texts are necessary. There are documented steps to increase to >1M tokens such as this [article](https://gradient.ai/blog/scaling-rotational-embeddings-for-long-context-language-models), which also involves increasing the rope theta while increasing length of the training sequence. This example training [script](https://github.com/jzhang38/EasyContext/blob/main/train_scripts/EasyContext-1M-Llama-2-7B.sh) might also be helpful.

Thanks for answering! That's pretty helpful for me!
